stf/src/sgx.rs

```diff
@@ -410,6 +449,35 @@ fn get_account_info(who: &AccountId) -> Option<AccountInfo> {
     }
 }
 
+fn validate_nonce(who: &AccountId, nonce: Index) -> bool {
+    // validate
+    let expected_nonce = get_account_info(who)
+        .map_or_else(|| 0, |acc| acc.nonce);
+    if expected_nonce == nonce {
+        return true
+    }
+    false
+}
+
+/// increment nonce after a successful call execution
+fn increment_nonce(account: &AccountId) {
```

Comment:
this fails silently. Is that good?

Reply:
It is not good. However, currently it needs to be allowed to fail for the cases described in the comments below (transactions of zero value and from to the same account). With the batch validation implementation and the checking of transfer fees, this function must not fail anymore.

Would you agree with the addition of a FIXME?

Reply:
Actually I was overthinking it. If it will fail later due to missing fees, why not now already due to nonce. Fixed with a simple unwrap.

Reply:
Note: Currently not a nice solution because worker will panic if a non-existent account successfully transfers 0 or to itself some money. This will be fixed with the fee check.

Reply:
I can live with a FIXME. I cannot live with a panic in the worker!

Reply:
Fixed accordingly